InternetHospital.BusinessLogic/Services/DoctorService.cs

```diff
@@ -248,5 +253,130 @@ private bool FillIllness(IllnessHistoryModel illnessModel, Appointment appointme
             return result;
         }
 
+        /// <summary>
+        /// Get all patients who had appointments for current doctor
+        /// </summary>
+        /// <param name="doctorId"></param>
+        /// <returns>
+        /// returns a collection of doctor's patients
+        /// </returns>
+        private IQueryable<User> GetDateOfMyPatients(int doctorId)
+        {
+            var patients = _context.Users.Where(x => x.Appointments.Any(a => a.DoctorId == doctorId))
+                .Select(p => new User
```

Comment:
You can use Automapper here.

Reply:
Yes, but I do not want to do this, he uses additional resources, but from him the code is much less readable. In addition, you twice (ModeratorService and ArticleService) did not use AutoMapper in a similar situation